ipld/amt/src/amt.rs

```diff
@@ -517,25 +518,16 @@ where
     where
         F: FnMut(u64, &V) -> anyhow::Result<()>,
     {
-        if let Some(start_at) = start_at {
-            if start_at >= nodes_for_height(self.bit_width(), self.height() + 1) {
-                return Ok((0, None));
+        let mut num_traversed = 0;
+        for kv in self.iter_from(start_at.unwrap_or(0))? {
+            let (k, v) = kv?;
+            if limit.map(|l| num_traversed >= l).unwrap_or(false) {
+                return Ok((num_traversed, Some(k)));
             }
```

Comment:
Because...

1. I may or may not have a limit and I didn't really want to have a sentinel max value.
2. `take(limit)` takes a usize, this function takes a u64, and I didn't want to break anything.
3. I needed to take the limit, plus 1 (to get the next index). And I needed a way to detect when I'm on the last one. So I would have needed something like:

```rust
let mut iter = self.iter_from(...);
for kv in iter.by_ref().take(limit) {
}
if let Some(next_idx) = iter.next() {
}
```

(basically, I wrote it that way, and then realized this way was simpler)